cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -100,19 +92,12 @@ IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # easily fix them...)
   #
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-dangling-else")
-  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-delete-non-virtual-dtor") # not harmless but needed for boost <1.50.0
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-long-long")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-newline-eof")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-unsupported-friend")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-unused-function")
-  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-unused-parameter") # mainly caused by BOOST
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-unused-private-field")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-unused-variable")
-
-  # suppress warnings in boost 1.56:
-  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-c++11-extensions")
-  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-c99-extensions")
-  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-variadic-macros")
```

Comment:
So the problem I see with distributing these `ENABLE_IF_SUPPORTED` checks to the configuration of the sub-libraries is (i) that it's going to much harder to find out where exactly a flag is set, and (ii) that you need to adjust the configuration of sub-libraries or external dependencies for the various compilers we support -- i.e., that the BOOST configuration suddenly has to have switches in it for GCC, CLANG, etc. I find that not overly elegant.

I don't know how to best solve this. Your approach may well be the best one can do. I just wanted to raise the issue.